### ISSUE TITLE
Sub-Calendar not displaying events. 

### ISSUE THREAD

**Issue by james7342** [DNNCommunity/DNN.Events#248]
## Describe the bug
Secondary module on a separate page with a Sub-Calender configured to pull from a main calendar. 

## To Reproduce
1. Have a Main Event Module on a page with Events. 
2. Create secondary page and add event module. 
3. In Edit Settings for Event module on the secondary page and add a Sub-Calender for Main Event module as selected. 
4. On secondary page module will display events. 
5. Restart Application Pool from PB/Server/Restart Application
6. Notice on Secondary Page No events are shown.
7. Edit Settings to check and don't make any changes but select "Update"
8. Module on secondary page will show events unit application is restarted.

## Expected behavior
Settings and events should be shown through Application Restart

## Screenshots
Working
![screenshot_871](https://github.com/DNNCommunity/DNN.Events/assets/27960723/aefc8a28-2307-479b-bf9b-1191ac273f45)

After "Restart of application"
![screenshot_872](https://github.com/DNNCommunity/DNN.Events/assets/27960723/9028f3f0-00da-4d73-81e7-f1db896189a3)

Edit Settings and only click "Update"
![screenshot_873](https://github.com/DNNCommunity/DNN.Events/assets/27960723/7da291e7-9e02-4cd3-a79e-3475155f8f55)

Events Displayed again. 
![screenshot_874](https://github.com/DNNCommunity/DNN.Events/assets/27960723/89add9c3-d36c-4324-ab67-6ee2c12781a4)

## Error log
Will attempt with debug Dll's to see if any errors are log, but nothing with build Dll's. 

## Additional context
I know there is a current issue with settings not being saved. This worked for the longest time and broke recently after trying to turn off ToolTips on the module. Which those settings didn't seem to save or show unitil a restart of the application also. So I think it's some what related to issue #231

**Comment by james7342** [comment]
Slight edit to "7. Edit Settings to check and don't make any changes but select "Update"". Ran into a condition where clicking "update" didn't fix it. I had to make a change, then click update and the events showed. The change I made was turning off ToolTip for Month view and updating. Which will not show in the edit settings if you go back in. I'll open a new defect to cover that.

Opened new issue #249

**Comment by james7342** [comment]
Debug DLL"s installed. Duplicated the issue and nothing written to admin log or Portal/_default/Logs/ after "Restart Application" or "Edit Settings/update"

**Comment by james7342** [comment]
I think I might have figured this one out. I still have some more testing but I think it's related #229. Looking in the DB I had these 2 values in the ModuleSettings with the Module ID of the module the page that had the issues. I deleted both of these from the table and so far so good after a "Restart Application"

ModuleCategoryIds | System.Collections.ArrayList
ModuleLocationIds | System.Collections.ArrayList

I found this by comparing the DB from a prior working state to today. In the working DB the values for these settings were -1.
